### ISSUE TITLE
Propose a new rule about conditions of control statements

### ISSUE THREAD

**Issue by kerryChen95** [eslint/eslint#9854]
[The airbnb's JS code guide about conditions of control statements](https://github.com/airbnb/javascript#control-statements) is good and general, but there is no ESLint rule to make it effective.

>  In case your control statement (if, while etc.) gets too long or exceeds the maximum line length, each (grouped) condition could be put into a new line. The logical operator should begin the line.
>
> Why? Requiring operators at the beginning of the line keeps the operators aligned and follows a pattern similar to method chaining. This also improves readability by making it easier to visually follow complex logic.

```javascript
// bad
if ((foo === 123 || bar === 'abc') && doesItLookGoodWhenItBecomesThatLong() && isThisReallyHappening()) {
  thing1();
}

// bad
if (foo === 123 &&
  bar === 'abc') {
  thing1();
}

// bad
if (foo === 123
  && bar === 'abc') {
  thing1();
}

// bad
if (
  foo === 123 &&
  bar === 'abc'
) {
  thing1();
}

// good
if (
  foo === 123
  && bar === 'abc'
) {
  thing1();
}

// good
if (
  (foo === 123 || bar === "abc")
  && doesItLookGoodWhenItBecomesThatLong()
  && isThisReallyHappening()
) {
  thing1();
}

// good
if (foo === 123 && bar === 'abc') {
  thing1();
}
```

**Comment by ljharb** [comment]
The `operator-linebreak` rule already enforces that the operators are in the right place in the airbnb config (just not yet released).

**Comment by kerryChen95** [comment]
Thanks, I found that [`operator-linebreak` was `off` at `eslint-config-airbnb-base` latest version `v12.1.0`](https://github.com/airbnb/javascript/blob/eslint-config-airbnb-base-v12.1.0/packages/eslint-config-airbnb-base/rules/style.js#L378), I tried to config this rule by myself.
